src/pybind/mgr/dashboard/CMakeLists.txt

```
@@ -127,3 +127,14 @@ if(WITH_TESTS)
   include(AddCephTest)
   add_tox_test(mgr-dashboard TOX_ENVS py3 lint check openapi-check)
 endif()
+
+install(DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}
+  DESTINATION ${CEPH_INSTALL_DATADIR}/mgr/dashboard
+  ${mgr_module_install_excludes}
+  REGEX "node_modules" EXCLUDE
+  REGEX "cypress.*" EXCLUDE
+  REGEX "\.coveragerc" EXCLUDE
+  REGEX "\.editorconfig" EXCLUDE
+  REGEX "\..*lintrc" EXCLUDE
+  REGEX "\.browserslistrc" EXCLUDE
+  REGEX "\.prettier*" EXCLUDE)
```

I remember @a16bitsysop mentioned it didn´t work but the following should match all dot files:
```suggestion
  PATTERN ".*" EXCLUDE
  )
```

---

`REGEX "\..+" EXCLUDE` Matched everything, not just dot files, `.*` is supposed to match everything.

---

One thing I do on alpine as well is delete the node_modules after build step is finished to make sure there is enough space to install Ceph in the fake root, if the node_modules was not used for tests could it be deleted after the angular app is built?

---

updated accordingly.

---

I tried the previous version on the alpine ci:
https://gitlab.alpinelinux.org/a16bitsysop/aports/-/jobs/433650/raw

the differences between previous version (16.2.5) are listed at the bottom.

It seems to remove the mgr telemetry and status modules.
Remove some tests and adds others.
Create an extra rook folder.
Create an extra dashboard folder and install src and tests now.

I redid the patch so it patches against Ceph 16.2.5 as there was a conflict with this and the patch it is superceeding somehow.  It should the same changes as the previous version.

---

I created a “pre” patch so these apply unchanged and downloaded new versions, the result looks the same:
https://gitlab.alpinelinux.org/a16bitsysop/aports/-/jobs/433764/raw

---

@a16bitsysop Duncan, thank you for testing the change!

- the "telemetry and status" change was an oversight. 
- the extra "rook" and "dashboard" directories was regressions.  i should not specify the subdirectory name when installing them
- "Remove some tests and adds others", i added "tests" to the exclude list. hopefully it helps.

---

Okay, that looks better:
https://gitlab.alpinelinux.org/a16bitsysop/aports/-/jobs/434105/

---

I didn’t test the ceph.spec and Debian build commit as they wouldn’t change anything when building in the alpine ci

---

I found one thing actually, as I am patching CMakelists.txt I get a .orig file which is now added:

`+usr/share/ceph/mgr/dashboard/CMakeLists.txt.orig`

could .orig files be excluded?

---

i'd prefer not adding it. as .orig and .rej files are created by `patch` when a patch fails to apply cleanly. this is a case needs human intervention. and in our daily development or in CI, this does not happen quite often. so i'd like to keep the exclude list a minimal set.

---

Okay